fmriprep/interfaces/images.py

```diff
@@ -370,47 +370,54 @@ def _run_interface(self, runtime):
         # Check affine information is valid
         valid_codes = (qform_code, sform_code) != (0, 0)
 
+        # Matching affines
+        matching_affines = np.allclose(img.get_qform(), img.get_sform())
+
         # Both okay -> do nothing, empty report
-        if valid_qform and valid_codes:
+        if valid_qform and valid_codes and matching_affines:
             self._results['out_file'] = self.inputs.in_file
             open(out_report, 'w').close()
             self._results['out_report'] = out_report
             return runtime
 
-        snippet = ''
         out_fname = fname_presuffix(self.inputs.in_file, suffix='_valid', newpath=runtime.cwd)
         self._results['out_file'] = out_fname
 
+        warning = False
+        warning_txt = ''
+        description = ''
         # Fix sform if needed
         if not valid_codes:
             # Nibabel derives a default LAS affine from the shape and zooms
             # Use scanner xform code to indicate no alignment has been done
             img.set_sform(img.affine, nb.nifti1.xform_codes['scanner'])
-            snippet += """\
-<h3 class="elem-title">WARNING - Invalid header</h3>
+            warning = True
+            warning_txt = 'Invalid header'
+            description += """\
 <p class="elem-desc">Input file does not have valid qform or sform matrix.
   A default, LAS-oriented affine has been constructed.
   A left-right flip may have occurred.
   Analyses of this dataset MAY BE INVALID.
 </p>
 """
-        if not valid_qform:
+        if not valid_qform and old_qform_code != 0:
+            if not warning:
+                warning = True
+                warning_txt = 'Invalid qform'
+            description += """\
+<p class="elem-desc">Input file does not have a valid qform matrix.</p>
+"""
+
+        if not (valid_qform and matching_affines):
```

Comment:
like before, if qform is not valid then it doesn't make much sense to see if it matches the sform (or, better said, you can't even check).

Reply:
This is a short-circuit. We want to set the qform in the two cases: `not valid_qform` and `not matching_affines`.

Reply:
Yes, I just wanted to point out the fact that we need to set `matching_affines` in a different way for this switch to work as expected.